skopt/optimizer/base.py

```diff
@@ -186,129 +172,81 @@ def base_minimize(func, dimensions, base_estimator,
     n_init_func_calls = len(x0) if y0 is None else 0
     n_total_init_calls = n_random_starts + n_init_func_calls
 
-    if n_calls <= 0:
-        raise ValueError("Expected `n_calls` > 0, got %d" % n_calls)
-
-    if n_random_starts < 0:
+    if n_calls < n_total_init_calls:
         raise ValueError(
-            "Expected `n_random_starts` >= 0, got %d" % n_random_starts)
+            "Expected `n_calls` >= %d, got %d" % (n_total_init_calls,
+                                                  n_calls))
 
     if n_random_starts == 0 and not x0:
-        raise ValueError("Either set `n_random_starts` > 0, or provide `x0`")
-
-    if n_calls < n_total_init_calls:
-        raise ValueError(
-            "Expected `n_calls` >= %d, got %d" % (n_total_init_calls, n_calls))
+        raise ValueError("Either set `n_random_starts` > 0,"
+                         " or provide `x0`")
 
     callbacks = check_callback(callback)
     if verbose:
         callbacks.append(VerboseCallback(
             n_init=n_init_func_calls, n_random=n_random_starts,
             n_total=n_calls))
 
-    if y0 is None and x0:
-        y0 = []
+    # User suggested points at which to evaluate the objective first
+    if x0 and y0 is None:
         for i, x in enumerate(x0):
-            y0.append(func(x))
-            curr_res = create_result(x0[:i + 1], y0, space, rng, specs)
+            fit_model = i == len(x0) - 1
+            y = func(x)
+            optimizer.tell(x, y, fit=fit_model)
```

Comment:
It seems to me that when `x0`, and `n_random_starts` are present, the first model is fit after the `len(x0)` calls to the function but before it was `len(x0) + n_random_starts` no?

That is what I had meant by "behaviour change"